pallets/subtensor/src/weights.rs

```diff
@@ -40,11 +54,36 @@ impl<T: Config> Pallet<T> {
             Error::<T>::WeightsCommitNotAllowed
         );
 
-        WeightCommits::<T>::insert(
-            netuid,
-            &who,
-            (commit_hash, Self::get_current_block_as_u64()),
-        );
+        // Check if the nonce is greater than the last processed nonce
+        let last_processed_nonce = Self::get_last_processed_nonce(netuid, &who);
+        ensure!(nonce > last_processed_nonce, Error::<T>::NonMonotonicNonce);
+
+        WeightCommits::<T>::try_mutate(netuid, &who, |commits| -> DispatchResult {
+            // Check if the nonce already exists
+            ensure!(!commits.contains_key(&nonce), Error::<T>::DuplicateNonce);
```

Comment:
Is this error useful? We accomplish the same thing, more generically, with the greater than latest nonce check. 

Also removes a few reads versus just one we do anyway.

Reply:
makes sense . removing